### PR TITLE
Typehint shard.py and set force=True for disconnecting from voice clients in AutoShardedClient.close

### PR DESCRIPTION
**Pull request by StockerMC**
## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/shard.py [hunk 1]
```diff
@@ -34,22 +35,29 @@
...
 __all__ = (
     'AutoShardedClient',
     'ShardInfo',
 )
 
-log = logging.getLogger(__name__)
+log: logging.Logger = logging.getLogger(__name__)
 
 class EventType:
     close = 0
```
Comment (bryanforbes) [inline discord/shard.py:63]: These should all be annotated as `ClassVar[int]`

File: discord/shard.py [hunk 2]
```diff
@@ -62,36 +70,36 @@ class EventType:
...
 
 class Shard:
-    def __init__(self, ws, client, queue_put):
-        self.ws = ws
-        self._client = client
-        self._dispatch = client.dispatch
-        self._queue_put = queue_put
-        self.loop = self._client.loop
-        self._disconnect = False
+    def __init__(self, ws: DiscordWebSocket, client: AutoShardedClient, queue_put: Callable[[Any], None]) -> None:
```
Comment (bryanforbes) [inline discord/shard.py:92]: It would be better if `queue_put` was typed as `Callable[[EventItem], None]`

File: discord/shard.py [hunk 3]
```diff
@@ -62,36 +70,36 @@ class EventType:
...
+        self.ws: DiscordWebSocket = ws
+        self._client: Client = client
+        self._dispatch: Callable[..., None] = client.dispatch
+        self._queue_put: Callable[[Any], None] = queue_put
```
Comment (bryanforbes) [inline discord/shard.py:96]: The same goes for `_queue_put` here

File: discord/shard.py [hunk 4]
```diff
@@ -101,25 +109,25 @@ def __init__(self, ws, client, queue_put):
         )
 
     @property
-    def id(self):
-        return self.ws.shard_id
+    def id(self) -> int:
+        return self.ws.shard_id # type: ignore
```
Comment (bryanforbes) [inline discord/shard.py:113]: Please add a comment to explain why this is ignored

**Review by bryanforbes — CHANGES_REQUESTED** [review]

File: discord/shard.py [hunk 5]
```diff
@@ -34,64 +35,71 @@
...
-    resume = 2
-    identify = 3
-    terminate = 4
-    clean_close = 5
+    close: ClassVar[int] = 0
+    reconnect: ClassVar[int] = 1
+    resume: ClassVar[int] = 2
+    identify: ClassVar[int] = 3
+    terminate: ClassVar[int] = 4
+    clean_close: ClassVar[int] = 5
```
Comment (Rapptz) [inline discord/shard.py:68]: Please don't.

File: discord/shard.py [hunk 6]
```diff
@@ -315,18 +327,18 @@ def __init__(self, *args, loop=None, **kwargs):
         self._connection._get_client = lambda: self
         self.__queue = asyncio.PriorityQueue()
 
-    def _get_websocket(self, guild_id=None, *, shard_id=None):
-        if shard_id is None:
+    def _get_websocket(self, guild_id: Optional[int] = None, *, shard_id: Optional[int] = None) -> DiscordWebSocket:
+        if shard_id is None and guild_id is not None and self.shard_count is not None:
```
Comment (Rapptz) [inline discord/shard.py:331]: `_get_websocket` is always called with these being set.

File: discord/shard.py [hunk 7]
```diff
@@ -355,11 +367,11 @@ def get_shard(self, shard_id):
             return ShardInfo(parent, self.shard_count)
 
     @property
-    def shards(self):
+    def shards(self) -> Mapping[int, ShardInfo]:
```
Comment (Rapptz) [inline discord/shard.py:370]:
```suggestion
    def shards(self) -> Dict[int, ShardInfo]:
```

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/shard.py (hunk 6 above)
Comment (StockerMC) [reply discord/shard.py]: Changed back to the original conditional and added a `# type: ignore` with the reason.

**Review by Rapptz — APPROVED** [review]

**Pull request merged**